src/ComunicaEngine.js

```diff
@@ -22,23 +23,38 @@ export default class ComunicaEngine {
    * Creates an asynchronous iterable of results for the given SPARQL query.
    */
   async* execute(sparql, source) {
-    if ((/^\s*(?:INSERT|DELETE)/i).test(sparql))
+    if ((/^\s*(?:INSERT|DELETE)/i).test(sparql)) {
       yield* this.executeUpdate(sparql, source);
-
-    // Load the sources if passed, the default sources otherwise
-    const sources = await (source ? this.parseSources(source) : this._sources);
-    if (sources.length !== 0) {
-      // Execute the query and yield the results
-      const queryResult = await this._engine.query(sparql, { sources, ...this._options });
-      yield* this.streamToAsyncIterable(queryResult.bindingsStream);
+    }
+    else {
+      // Load the sources if passed, the default sources otherwise
+      const sources = await (source ? this.parseSources(source) : this._sources);
+      if (sources.length !== 0) {
+        // Execute the query and yield the results
+        const queryResult = await this._engine.query(sparql, { sources, ...this._options });
+        yield* this.streamToAsyncIterable(queryResult.bindingsStream);
+      }
     }
   }
 
   /**
    * Creates an asynchronous iterable with the results of the SPARQL UPDATE query.
    */
   async* executeUpdate(sparql, source) {
-    throw new Error(`SPARQL UPDATE queries are unsupported, received: ${sparql}`);
+    // Load the sources if passed, the default sources otherwise
+    let sources = await (source ? this.parseSources(source) : this._destination);
+    if (sources.length !== 0) {
+      // Update queries can only handle a single source
+      if (sources.length > 1)
+        sources = [sources[0]];
+      // Execute the query and yield the results
+      const queryResult = await this._engine.query(sparql, { sources, ...this._options });
+      if (queryResult.type !== 'update')
+        throw new Error(`Update query returned unexpected result type: ${queryResult.type}`);
+
+      // Resolves when the update is complete
+      return queryResult.updateResult;
```

Comment:
Hmm, is that the result we want though?

Reply:
`updateResult` is of type `Promise<void>` so I was under the impression this would have the same effect as
```ts
await queryResult.updateResult;
return;
```

I can add some tests to confirm that is the case

Reply:
It's rather that the original code higher up (originally without the `else` statement) would return the updated results.

Reply:
I don't quite follow what you mean my this.

If it is an INSERT/DELETE query then this should just wait until `updateResult` resolves, and since it resolves to `void`, this should not cause anything to be yielded/returned and so the Iterator will end.

If it is a `SELECT` query then it should iterate through the bindings as expected due to the `yield*`

Reply:
Yeah; see my comment there. Basically can't remember why there was no `else` in the original code.

Reply:
The else statement was added because originally with INSERT/DELETE queries an error was thrown so the `yield*` now inside the else statement was never reached on INSERT/DELETE queries.